### ISSUE TITLE
Code coloring of filter blocks breaks when there are empty lines

### ISSUE THREAD

**Issue by bensaufley** [karuna/haml-vscode#3]
The following is valid HAML:

```haml
:ruby
  my_var = "blah"
  # No spaces on the below line

  another_var = "blarg"

%div
```

Of course in this limited example, that extra line between `my_var` and `another_var` seems superfluous. But for larger blocks of Ruby (or other code), it makes sense to break up bits of the code. Ideally, I'm not putting large blocks of Ruby in the middle of my HAML, but … sometimes it must be done.

Anyway: the current configuration treats that empty line as the end of the `:ruby` block, and the following lines are just rendered in white text:

<img width="306" alt="screen shot 2017-08-30 at 10 37 36 am" src="https://user-images.githubusercontent.com/331287/29878008-53fb3626-8d6f-11e7-8534-290e14d46a7e.png">

If I were to add indentation (in this case, two spaces, to match the other lines), it would behave as expected:

<img width="311" alt="screen shot 2017-08-30 at 10 38 46 am" src="https://user-images.githubusercontent.com/331287/29878037-702208de-8d6f-11e7-9645-9f6b916e963f.png">

But my code editor and my Rubocop config both don't like that – the former automatically removes trailing whitespace; the latter complains if I manage to include it.

**tl;dr**: If I have an empty line without indentation between lines in a `:filter` block, I **expect** code coloring to continue to the next line (second screenshot above); I **experience** white text instead (first screenshot)

**Comment by karuna** [comment]
@bensaufley I will look into that soon
